### DISCUSSION TITLE
Best scripts / scanners / thirdparty discovery agent?

### DISCUSSION THREAD

**Discussion by spastro** [snipe/snipe-it#11918] [Q&A]
Just deployed, sadly i did not realise there was no agent. 
Any activerly supported scripts / agents worth investigating to pull in as much data as i can for devices other than a csv?

Cheers

**Comment by snipe** [comment]
https://github.com/snipe/snipe-it#libraries-modules--related-projects

**Comment by michaeljhill** [comment]
Adding one more option for folks who land here looking specifically for an endpoint-side Snipe-IT agent:

https://github.com/itmik-software/snipe-it-agent

It is an independent Windows/macOS agent for syncing selected computer inventory into Snipe-IT through the REST API. Current focus is hardware asset creation/update, serial/model/manufacturer resolution, custom field mappings, optional installed-software capture, user assignment, dry-run testing, Windows service deployment, Windows Intune packaging, and macOS/Intune deployment workflows.

Not an official Snipe-IT project, and not an RMM/patching/compliance platform. I am maintaining it donation-supported mostly to fund signing, packaging, docs, and compatibility testing. Feedback from anyone solving the same “keep Snipe-IT computers current without manual entry” problem would be very welcome.
